test/lib/interpreters/interpreter_test.rb

```diff
@@ -19,719 +19,128 @@
 require 'date'
 
 class InterpreterTest < Minitest::Test
-  def test_expand_vrp_sequence_and_visit_range
-    size = 5
-    problem = {
-      matrices: [{
-        id: 'matrix_0',
-        time: [
-          [0,  1,  1,  10, 0],
-          [1,  0,  1,  10, 1],
-          [1,  1,  0,  10, 1],
-          [10, 10, 10, 0,  10],
-          [0,  1,  1,  10, 0]
-        ],
-        distance: [
-          [0,  1,  1,  10, 0],
-          [1,  0,  1,  10, 1],
-          [1,  1,  0,  10, 1],
-          [10, 10, 10, 0,  10],
-          [0,  1,  1,  10, 0]
-        ]
-      }],
-      points: (0..(size - 1)).collect{ |i|
-        {
-          id: "point_#{i}",
-          matrix_index: i
-        }
-      },
-      rests: [{
-        id: 'rest_0',
-        timewindows: [{
-          day_index: 0,
-          start: 1,
-          end: 1
-        }],
-        duration: 1
-      }],
-      vehicles: [{
-        id: 'vehicle_0',
-        start_point_id: 'point_0',
-        matrix_id: 'matrix_0',
-        rest_ids: ['rest_0'],
-        sequence_timewindows: [{
-          start: 1,
-          end: 1
-        }]
-      }],
-      services: (1..(size - 1)).collect{ |i|
-        {
-          id: "service_#{i}",
-          activity: {
-            point_id: "point_#{i}",
-            timewindows: [{
-              start: 1,
-              end: 2
-            }, {
-              start: 5,
-              end: 7
-            }]
-          },
-          visits_number: 2
-        }
-      },
-      configuration: {
-        preprocessing: {
-          cluster_threshold: 5
-        },
-        resolution: {
-          duration: 10
-        },
-        schedule: {
-          range_indices: {
-            start: 0,
-            end: 1
-          }
-        }
-      }
-    }
+  def periodic_expand(problem)
     vrp = TestHelper.create(problem)
     periodic = Interpreters::PeriodicVisits.new(vrp)
-    expanded_vrp = periodic.send(:expand, vrp, nil)
-    assert_equal 2, expanded_vrp[:vehicles].size
-    assert_equal 2 * (size - 1), expanded_vrp[:services].size
-    assert(expanded_vrp[:services].all?{ |service| service.activity.timewindows.size == 2 })
-    assert_equal expanded_vrp[:services][0].activity.timewindows[0][:start], expanded_vrp[:services][1].activity.timewindows[0][:start]
+    periodic.send(:expand, vrp, nil)
```

Comment:
Interpreters::PeriodicVisits is initialized using the vrp, then we send the vrp a second time, cant we just send it a single time ?

Reply:
You are right, it does not seem so logical to send vrp twice.

However this looks like a big change in periodic_visits.rb. I suggest that we create an issue for this.